src/check-issue-labels.ts

```diff
@@ -0,0 +1,80 @@
+import {type Octokit} from '@octokit/core'
+import * as core from '@actions/core'
+import {checkLinkedIssueLabels} from './check-linked-issue-labels'
+
+export interface ClosingIssueReferenceEdge {
+  node: {
+    labels: {
+      edges: {
+        node: {
+          name: string
+        }
+      }[]
+    }
+    number: number
+  }
+}
+
+export interface QueryResult {
+  repository: {
+    pullRequest: {
+      closingIssuesReferences: {
+        edges: ClosingIssueReferenceEdge[]
+      }
+    }
+  }
+}
+
+export interface PullLocator {
+  owner: string
+  pull: number
+  repo: string
+}
+
+export async function checkIssueLabels(
+  client: Octokit,
+  locator: PullLocator,
+  requiredLabels: string[]
+) {
+  if (!requiredLabels.length) {
+    return []
+  }
+
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    const result = (await client.graphql(
```

Comment:
i get the feeling that this use of API and data exposure could bring on some new permissions requirements. Before this, we had just been looking mostly at what was on the payload, and the permissions came into play for commenting on issues and such (I suppose there was some with the ignore-team-members thing or whatever it's called). if you can double check API docs on the GraphQL side and see how those might translate to the permissions ramifications on the workflow file.

Reply:
Ok I poked around and don't see anything different permissions-wise. https://github.com/JoshuaKGoldberg/testing-issue-label-action/pull/2 shows it working with a basic actions setup (added to the PR description). So I hope there's nothing to do here?